scio-core/src/main/scala/com/spotify/scio/coders/KryoAtomicCoder.scala

```diff
@@ -79,162 +83,169 @@ private[scio] class KryoAtomicCoder[T](private val options: KryoOptions) extends
 
   import KryoAtomicCoder.logger
 
-  private val kryoState: ThreadLocal[KryoState] =
-    new EmptyOnDeserializationThreadLocal[KryoState] {
-      override def initialValue(): KryoState = {
+  @transient private lazy val kryoPool: KryoPool = {
```

Comment:
Please correct me if I'm wrong but this will possibly create multiple KryoPool for each instance of `KryoAtomicCoder`, if pooling KryoPool is thread safe, why not move this to an object and remove `@transient lazy`?

What's the footprint of this change?

Reply:
@ravwojdyla I was talking to @nevillelyh to clarify a few things and yes I guess this is not ideal. The main reason why we can't move it easily to an object is because we depend on KryoOptions. I guess we need to spend more time on this ...

Reply:
@ravwojdyla scratch that ... I think i have something ...

Reply:
@ravwojdyla updated wdyt?